### ISSUE TITLE
Prisma2 not working with netlify-lambda

### ISSUE THREAD

**Issue by seawatts** [prisma/prisma#1015]
I am using the netlify lambda cli. Not the netlify cli. https://github.com/netlify/netlify-lambda

Following up from the discussion here: https://github.com/prisma/prisma2/issues/952

When I try to run prisma2 generate I get this error

```
Error: The `platforms` field on the generator definition is deprecated. Please rename it to `binaryTargets`.
```

When I switch platforms to binaryTargets it runs generate fine. But I still get this error on invocation

```
Request from ::1: GET /.netlify/functions/hello
Response with status 500 in 528 ms.
Error during invocation:  Error: Error in Photon: 
Error: Photon binary for current platform darwin could not be found.
Photon looked in null but couldn't find it.
Make sure to adjust the generator configuration in the schema.prisma file:
generator photon {
  provider      = "photonjs"
  binaryTargets = ["native", "rhel-openssl-1.0.x", "darwin"]
}
Please run prisma2 generate for your changes to take effect.
Note, that by providing `native`, Photon automatically resolves `darwin`.
Read more about deploying Photon: https://github.com/prisma/prisma2/blob/master/docs/core/generators/photonjs.md
    at NodeEngine.getPrismaPath (~/bot-slack/lambda/hello.js:1:894920)
    at async ~/bot-slack/lambda/hello.js:1:896510
    at u.<anonymous> (~/bot-slack/lambda/hello.js:1:2379)
    at Generator.throw (<anonymous>)
    at u (~/bot-slack/lambda/hello.js:1:1380)
```

Here is my generator section of my schema.prisma

```
generator photon {
  provider = "photonjs"
  binaryTargets = ["native", "rhel-openssl-1.0.x"]
}
```

Even if I try adding darwin to that list it still errors, and just adds another darwin to the error message.

Make sure to adjust the generator configuration in the schema.prisma file:

```
generator photon {
  provider      = "photonjs"
  binaryTargets = ["native", "rhel-openssl-1.0.x", "darwin", "darwin"]
}
```

**Comment by pantharshit00** [comment]
I am able to reproduce this. This is related to how photon should be handled with bundlers. 

![image](https://user-images.githubusercontent.com/22195362/70076849-ac632080-1625-11ea-8a8c-30aed82eebcf.png)


We already have issues on that: https://github.com/prisma/photonjs/issues/119
https://github.com/prisma/specs/issues/336

Marking as a bug for now

**Comment by heymartinadams** [comment]
@seawatts facing the same issue over here. Commented also [here](https://github.com/netlify/zip-it-and-ship-it/issues/81).

**Comment by pantharshit00** [comment]
@heymartinadams Try using the following guide: https://prisma2.netlify.com/guides/deployment/deploying-to-netlify

**Comment by heymartinadams** [comment]
Able to deploy, but functions not working: https://github.com/heymartinadams/deployment-example-netlify

```
11:33:50 PM: 2020-03-31T06:33:50.347Z	b4858c2f-d078-4eaa-be48-10935f847f02	ERROR	PrismaClientKnownRequestError: 
Invalid `prisma.user.findMany()` invocation in
/var/task/src/functions/users.js:6:37

Can't reach database server at `SG-LifePurposeApp-524-pgsql-master.servers.mongodirector.com`:`5432`

Please make sure your database server is running at `SG-LifePurposeApp-524-pgsql-master.servers.mongodirector.com`:`5432`.
    at PrismaClientFetcher.request (/var/task/src/node_modules/@prisma/client/runtime/index.js:1:47358)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  code: 'P1001',
  meta: {
    database_host: 'SG-LifePurposeApp-524-pgsql-master.servers.mongodirector.com',
    database_port: 5432
  }
}
```

**Comment by pantharshit00** [comment]
@heymartinadams Error suggests the Prisma wasn't able to reach your database. Check your database connection string.

**Comment by heymartinadams** [comment]
@pantharshit00 Fixed the string, but still not working 😕 Will need to ask for help. Might you (or someone you know) be available to assist (for compensation, of course)?

**Comment by heymartinadams** [comment]
Unable to get Prisma Client binary copied over in Netlify Functions

I followed the exact instructions provided in the [docs](https://www.prisma.io/docs/guides/deployment/deploying-to-netlify), and got the [example repo](https://github.com/prisma/deployment-example-netlify) to work.

However, my project uses the `netlify-lambda` command during the build process (the demo doesn’t use `netlify-lambda`); somehow the Prisma Client binary isn’t being copied over during the install process. Here’s the error I’m receiving.

```
Invalid `prisma.user.findOne()` invocation:

Query engine binary for current platform "rhel-openssl-1.0.x" could not be found.
This probably happens, because you built Prisma Client on a different platform.
(Prisma Client looked in "/query-engine-rhel-openssl-1.0.x")

You already added the platforms "native", "rhel-openssl-1.0.x", "debian-openssl-1.0.x" to the "generator" block
in the "schema.prisma" file as described in https://pris.ly/d/client-generator,
but something went wrong. That's suboptimal.

Please create an issue at https://github.com/prisma/prisma-client-js/issues/new
    at A.request (/var/task/src/build-functions/graphql.js:178:102125)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

Tried everything for days. Not sure if there’s an error in my build process (my code used in the build itself works fine: app is running, just not Prisma Client), or if the error lies with Netlify’s Prisma integration.

## netlify.toml

```toml
[build]
publish = "build/"
# Needed to take all netlify-lambda commands out of the package.json file and place them directly into the netlify.toml file for production, since during development Netlify attempts to run scripts that contain `netlify-lambda` functions! See: https://github.com/netlify/cli/issues/723#issuecomment-605511147
command = "cd functions && yarn && cd .. && npx prisma2 generate && netlify-lambda build functions && netlify-lambda install functions && react-scripts build"
# During production, `functions` folder cannot be the same as destination, so you need to create another destination.
functions = "build-functions/"

[build.environment]
# Fix to ensure the Prisma binary is packaged with the lambda function
ZISI_VERSION = "0.4.0-9"

[context.develop]
# `netlify dev` in local development won’t work unless destination is the same as actual `functions` folder.
functions = "functions/"
```

**Comment by janpio** [comment]
Please open a new, separate issue @heymartinadams - then someone will try to help you.

**Comment by janpio** [comment]
Internal Note: Opened issue for Netlify, waiting for progress there.

**Comment by janpio** [comment]
Issue at Netlify side has been auto closed :(

**Comment by pantharshit00** [comment]
I am not sure people are even using that package. Seems like netlify dev is the new cli now.

These multiple CLIs from netlify is confusing.....

**Comment by janpio** [comment]
Reached out internally: https://prisma-company.slack.com/archives/CAZ0TJPFV/p1616782683004000

**Comment by verythorough** [comment]
Hello from Netlify! In the years since `netlify-lambda` was created, we've since released [direct bundling and shipping of JavaScript functions](https://docs.netlify.com/functions/build-with-javascript/#unbundled-javascript-function-deploys), so you no longer need to add a step to build your functions using `netlify-lambda`.

Also, as @pantharshit00 noted, we added the `dev` command to Netlify CLI, which serves JS functions directly, again with no need for a separate build tool. (The [docs](https://docs.netlify.com/functions/build-with-javascript/#tools) reflect these recommendations as well.)

The one remaining functionality that `netlify-lambda` provides is webpack bundling. We're currently working on expanding our functions bundler options directly in the build, so we'll soon be able to replace that functionality, too. When _that_ happens, we'll be able to archive `netlify-lambda`.
